### ISSUE TITLE
Convert needed between numpy.ma.core.MaskedArray() and astropy.nddata.CCDData() 

### ISSUE THREAD

**Issue by janerigby** [astropy/astropy#11237]
Because the astropy-affiliated package ccdproc does not have all the clipping parameters of astropy.stats.sigma_clip() (see issue #753)  I'm having to make the mask using sigma_clip. The output of sigma_clip() is  numpy.ma.core.MaskedArray().  But then ccdproc expects the astropy.nddata.CCDData() class.  There doesn't seem to be a converter between numpy.ma.core.MaskedArray() and CCDData; it seems like one would be very useful.

I think it could be as trivial as:
`CCDData(clipped.data, mask=clipped.mask, unit='adu') `
where clipped is the masked array output of sigma_clip. 
Alternatively, ccdproc could just handle masked arrays.
A third approach would be for sigma_clip to optionally output CCDData objects.

**Comment by pllim** [comment]
Thanks for reporting this! I think the first proposed solution makes the most sense, but let's see what @mwcraig says.

**Comment by dhomeier** [comment]
Wouldn't the first suggestion still require `CCDData` to somehow handle `MaskedArray`, or how would you want it to treat the masked values? Or should it replace them similar to e.g.
`CCDData(clipped.data.filled(-99.9), unit='adu')` ?

**Comment by saimn** [comment]
A `CCDData` object can already be instantiated from a Numpy masked array (e.g. `CCDData(clipped, unit='adu')`).

**Comment by janerigby** [comment]
I agree, and I stumbled upon that solution*.   It would be helpful if the CCDData documentation mentioned the existence of numpy masked arrays, and suggested the best-practice conversion.

* (see above, although I explicitly sent it the mask=, because I wasn't sure that otherwise CCDData would import the mask.  This is where documentation would help.)
